lib/engine/game/g_1894/step/track.rb

```diff
@@ -15,7 +15,12 @@ def actions(entity)
             super
           end
 
-          def legal_tile_rotation?(_entity, hex, tile)
+          def legal_tile_rotation?(entity, hex, tile)
+            if @game.class::GREEN_CITY_TILES.include?(hex.tile.name)
+              entity_reaches_a_new_exit = !(tile.exits & hex_neighbors(entity, hex)).empty?
+              return false unless entity_reaches_a_new_exit
+            end
```

Comment:
The logic here looks correct. The only thing I would change is to use a slightly more efficient way for testing for array intersection: `arr1.intersect?(arr2)` instead of `(arr1 & arr2).empty?`. You could also stick it as a single liner, but Rubocop might object to the line length.
```suggestion
            return false if @game.class::GREEN_CITY_TILES.include?(hex.tile.name) && !tile.exits.intersect?(hex_neighbors(entity, hex))
```